pkg/scheduler/scheduler.go

```diff
@@ -324,23 +324,51 @@ type partialAssignment struct {
 	preemptionTargets []*workload.Info
 }
 
+// limitUsage makes sure that cluster queue usage does not exceed nominal+borrowing quota
+func limitUsage(e flavorassigner.Assignment, cq *cache.ClusterQueue) flavorassigner.Assignment {
+	for flavor, resourceUsage := range e.Usage {
+		cqQuota := make(map[corev1.ResourceName]*cache.ResourceQuota)
+		for _, rg := range cq.ResourceGroups {
+			for _, cqFlavor := range rg.Flavors {
+				if cqFlavor.Name == flavor {
+					cqQuota = cqFlavor.Resources
+				}
+			}
+		}
+		for resource, usage := range resourceUsage {
+			if cqQuota[resource].BorrowingLimit != nil {
+				usage = min(usage, cqQuota[resource].Nominal+*cqQuota[resource].BorrowingLimit-cq.Usage[flavor][resource])
+			}
+			resourceUsage[resource] = usage
+		}
+		e.Usage[flavor] = resourceUsage
+	}
+	return e
+}
+
 func (s *Scheduler) getAssignments(log logr.Logger, wl *workload.Info, snap *cache.Snapshot) (flavorassigner.Assignment, []*workload.Info) {
 	cq := snap.ClusterQueues[wl.ClusterQueue]
 	fullAssignment := flavorassigner.AssignFlavors(log, wl, snap.ResourceFlavors, cq, nil)
-	var fullAssignmentTargets []*workload.Info
+	var preemtionTargets []*workload.Info
```

Comment:
The intention of the old name was to differentiate from the preemptionTargets when there is partial admission. Maybe we can name it `faPreemptionTargets`